checks/shell_download_validate.go

```diff
@@ -493,91 +507,91 @@ func validateCommandIsNotNotSuFetchToStdinExecute(cmd, pathfn string,
 	return cmdValidated, nil
 }
 
-func isSuFetchStdinExecute(node syntax.Node, cmd, pathfn string,
+func validateCommandIsNotFileExecute(cmd, pathfn string, files map[string]bool,
 	logf func(s string, f ...interface{})) (bool, error) {
+	in := strings.NewReader(cmd)
+	f, err := syntax.NewParser().Parse(in, "")
+	if err != nil {
+		return false, ErrParsingShellCommand
+	}
+
+	cmdValidated := true
+	syntax.Walk(f, func(node syntax.Node) bool {
+		// Check if we're calling a file we previously downloaded.
+		if isExecuteFiles(node, cmd, pathfn, files, logf) {
+			cmdValidated = false
+		}
+
+		// Continue walking the node graph.
+		return true
+	})
+
+	return cmdValidated, nil
+}
+
+func extractInterpreterCommandFromNode(node syntax.Node) (string, bool) {
 	ce, ok := node.(*syntax.CallExpr)
 	if !ok {
-		return false, nil
+		return "", false
 	}
 
 	c, ok := extractCommand(ce)
 	if !ok {
-		return false, nil
+		return "", false
 	}
 
-	if !isSuCommand(c) {
-		return false, nil
+	if !isInterpreterWithCommand(c) {
+		return "", false
 	}
 
-	cs, ok := extractSuCommand(ce.Args)
+	cs, ok := extractInterpreterCommandFromArgs(ce.Args)
 	if !ok {
-		return false, nil
-	}
-
-	// We have extracted the command. Now parse it.
-	in := strings.NewReader(cs)
-	f, err := syntax.NewParser().Parse(in, "")
-	if err != nil {
-		return false, fmt.Errorf("syntax.NewParser: %w", err)
+		return "", false
 	}
 
-	isFetch := false
-	syntax.Walk(f, func(node syntax.Node) bool {
-		if isFetchStdinExecute(node, cmd, pathfn, logf) {
-			isFetch = true
-		}
-
-		// Continue walking the node graph.
-		return true
-	})
-
-	return isFetch, nil
+	return cs, true
 }
 
-func validateCommandIsNotSuFetchToStdinExecute(cmd, pathfn string,
-	logf func(s string, f ...interface{})) (bool, error) {
+func extractInterpreterCommandFromString(cmd string) (c string, res bool, err error) {
 	in := strings.NewReader(cmd)
 	f, err := syntax.NewParser().Parse(in, "")
 	if err != nil {
-		return false, ErrParsingShellCommand
+		return "", false, ErrParsingShellCommand
 	}
 
-	cmdValidated := true
+	cs := ""
+	ok := false
 	syntax.Walk(f, func(node syntax.Node) bool {
-		if err != nil {
+		if ok {
 			return false
 		}
-		var b bool
-		b, err = isSuFetchStdinExecute(node, cmd, pathfn, logf)
-		if err != nil {
+		s, kk := extractInterpreterCommandFromNode(node)
+		// nolinter
+		if kk {
```

Comment:
These variables (kk, cs, s) make this pretty confusing to understand. Can you rename them to more descriptive ones?

Reply:
sure. I wanted to just do `cs, ok = extractInterpreterCommandFromNode(node)` but it does not work, ie `cs` seems to be a local alias, so I had to create these new variables.

Reply:
renamed to `command, commandFound := extractInterpreterCommandFromNode(node)`